godot-core/src/obj/script.rs

```diff
@@ -360,9 +360,11 @@ where
         return false;
     }
 
-    let object_script: Gd<Script> = object_script_variant.to();
-
-    if object_script.upcast_ref::<Script>().__object_ptr() != script.upcast_ref().__object_ptr() {
+    if object_script_variant
+        .object_id()
+        .map(|instance_id| instance_id != script.instance_id())
+        .unwrap_or(true)
+    {
```

Comment:
It looks like this changes behavior:

previous condition was `a != b`
so if `a` is null, the overall condition is only true if `b` is *not* null.

now it's `a.map(|a| a != b).unwrap_or(true)`
meaning if `a` is null, the whole condition is always true.

Reply:
btw, this could be [`Option::map_or()`](https://doc.rust-lang.org/std/option/enum.Option.html#method.map_or), no?

Reply:
> previous condition was a != b
> so if a is null, the overall condition is only true if b is not null.
> 
> now it's a.map(|a| a != b).unwrap_or(true)
> meaning if a is null, the whole condition is always true.

B is always not null. A _should_ never be null, but we have no guarantee for that. Previously the `Variant::to` would panic if it was null, now the function fails. We also check if a is null before accessing the object_id so it's very unlikely that we will run into the `None` case.

> btw, this could be [Option::map_or()](https://doc.rust-lang.org/std/option/enum.Option.html#method.map_or), no?

Right, I will use that.